apps/entry/widgets/number_matrix_widget.py

```diff
@@ -29,3 +29,38 @@ def update_attribute(widget, _data, widget_data):
             }
         }
     }
+
+
+def get_comprehensive_data(widgets_meta, widget, _data, widget_data):
+    data = (_data or {}).get('value') or {}
+
+    if widgets_meta.get(widget.pk) is None:
+        row_headers_map = {
+            row['key']: row
+            for row in widget_data.get('row_headers', [])
+        }
+        column_headers_map = {
+            col['key']: col
+            for col in widget_data.get('column_headers', [])
+        }
+        widgets_meta[widget.pk] = {
```

Comment:
Maybe widgets_meta should not be mutated? It might affect the value at the calling point.

Reply:
It's expected behavior. Widget handles their cache data their self.